### PR TITLE
fix: Vite dev server add searchForWorkspaceRoot

### LINKED ISSUE #26035: Vite server not working in monorepo
### Current behavior

Project root is not recognised in monorepo (Vite dev server).

### Desired behavior

Work correctly in monorepo.

Add `searchForWorkspaceRoot(process.cwd())` to https://github.com/cypress-io/cypress/blob/434e6eec6e54461041774059b2d4ef5143b56a3d/npm/vite-dev-server/src/resolveConfig.ts#L114

### Test code to reproduce

Similar to https://stackoverflow.com/questions/74902697/error-the-request-url-is-outside-of-vite-serving-allow-list-after-git-init

### Cypress Version

12.7.0

### Node version

18.13.0

### Operating System

macOO

### Debug Logs

_No response_

### Other

Vite docs: https://vitejs.dev/config/server-options.html#server-fs-allow

### PR DESCRIPTION
**Pull request by radoslavirha**
- Closes https://github.com/cypress-io/cypress/issues/26035#issue-1612899517

### Additional details
Because `server.fs.allow` is defined, I need to detect workspace root correctly.

https://vitejs.dev/config/server-options.html#server-fs-allow

### Steps to test
Create monorepo with Vite app and test with Cypress.

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Review by lmiller1990 — APPROVED** [review]

**Comment by lmiller1990** [comment]
Nice, thanks a lot! We will get one more ✅ and this can go in the next release.

File: npm/vite-dev-server/src/resolveConfig.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@
  * You can find it here https://github.com/vitest-dev/vitest/blob/main/packages/vitest/src/node/create.ts
  */
 import debugFn from 'debug'
+import { searchForWorkspaceRoot } from 'vite'
```
Comment (lmiller1990) [inline npm/vite-dev-server/src/resolveConfig.ts]: Looks like this is causing CI to fail - I think the issue here is we don't bundle `vite` - that's from userland. So during the build step for Cypress, this doesn't actually exist.

We have something similar landing https://github.com/cypress-io/cypress/pull/26066/files#diff-5617aebe2ce067468c76a778abff48d659d1a28b47e93b2d0fff69f27cb2119eR61

We can either:

1. just copy paste from vite: https://github.com/vitejs/vite/blob/6b40f03574cd71a17cbe564bc63adebb156ff06e/packages/vite/src/node/server/searchRoot.ts#L59
2. create a npm package that does this? might be useful. Then ship it with the Cypress app
3. source the user's vite config and load via dynamic import, eg

```ts
export const createViteDevServerConfig = async (config: ViteDevServerConfig, vite: Vite): Promise<InlineConfig> => {
  const vite = await import('vite')
  const { findWorkspaceRoot } = vite
```

I think (2) or (3) is good. Thoughts?

**Review by lmiller1990 — COMMENTED** [review]
Hi, I realized this won't work as is, but can be fixed with a small change, see comment, thanks!

File: npm/vite-dev-server/src/resolveConfig.ts (hunk 1 above)
Comment (radoslavirha) [reply npm/vite-dev-server/src/resolveConfig.ts]: @lmiller1990 what is the second argument passed to `createViteDevServerConfig`? Is it Vite instance itself?

I'm pretty lost here on how cypress works so I'm wondering if you can't use the `findWorkspaceRoot` from `vite` argument

Comment (lmiller1990) [reply npm/vite-dev-server/src/resolveConfig.ts]: Huh, I think you are correct - nice catch, it looks like we already solved this at some point (been a while, I forgot how this works exactly).

We do have access to the user's vite, see here: https://github.com/cypress-io/cypress/blob/9c98b5e08583764ef228d9b8a40e5f28e7f41ade/npm/vite-dev-server/src/resolveConfig.ts#L54

Go for it, want to update this and do `vite.findWorkspaceRoot` using the Vite instance we receive? That should work.

**Review by lmiller1990 — APPROVED** [review]

**Comment by lmiller1990** [comment]
Nice - @mike-plummer can you take a quick 👀 , too?

**Review by mike-plummer — APPROVED** [review]
LGTM, will merge this later today

**Review by jordanpowell88 — APPROVED** [review]

**Comment by lmiller1990** [comment]
We've actually got a real failure now, we may need to rework one test: https://app.circleci.com/pipelines/github/cypress-io/cypress/50439/workflows/06809a66-8100-450e-a443-61812ef1d6d0/jobs/2094959/tests#failed-test-0

I think it's now detecting the vite version in the main monorepo, not the stale version the test is supposed to be detecting. I don't expect this will happen in production, it's a side effect of us running tests against the non-built Cypress binary.

Maybe someone internally can pick this up? I will try and take a look.

**Comment by lmiller1990** [comment]
Ahh, this API is new, we just need a null check. It's in v3 and v4.

image

**Comment by radoslavirha** [comment]
Thank you @lmiller1990 !

**Comment by lmiller1990** [comment]
Yep, pls wait around 2 weeks for the next release.

**Pull request merged**